src/mmw/apps/modeling/tests.py

```diff
@@ -1056,108 +1076,132 @@ def test_weather_post_project_owner_can_post(self):
+        scenario = self.create_private_scenario_with_weather_data()
 
-        response = self.c.post(self.endpoint(project_id), {})
+        response = self.c.post(self.endpoint(scenario['id']), {})
 
         self.assertEqual(response.status_code, 400)
 
     def test_weather_post_invalid_weather_400s(self):
-        project_id = self.create_private_project_with_weather_data()
+        scenario = self.create_private_scenario_with_weather_data()
 
-        response = self.c.post(self.endpoint(project_id),
+        response = self.c.post(self.endpoint(scenario['id']),
                                {'XYZ': self.weather_data_file},
                                format='multipart')
 
         self.assertEqual(response.status_code, 400)
 
-    # PATCH
-    def test_weather_patch_project_owner_can_patch(self):
-        project_id = self.create_private_project_with_weather_data()
+    # PUT
+    def test_weather_put_project_owner_can(self):
+        scenario = self.create_private_scenario_with_weather_data()
+        scenario['weather_type'] = WeatherType.DEFAULT
 
-        response = self.c.patch(self.endpoint(project_id),
-                                {'uses_custom_weather': False},
-                                format='json')
+        response = self.c.put('/mmw/modeling/scenarios/{}'
+                              .format(scenario['id']),
+                              scenario,
+                              format='json')
 
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.data['uses_custom_weather'], False)
-        self.assertNotEqual(response.data['custom_weather_dataset'], None)
+        self.assertEqual(response.data['weather_type'], WeatherType.DEFAULT)
+        self.assertNotEqual(response.data['weather_custom'], None)
 
-    def test_weather_patch_logged_out_user_cant_patch(self):
-        project_id = self.create_private_project_with_weather_data()
+    def test_weather_put_logged_out_user_cant(self):
+        scenario = self.create_private_scenario_with_weather_data()
+        scenario['weather_type'] = WeatherType.DEFAULT
         self.c.logout()
 
-        response = self.c.patch(self.endpoint(project_id),
-                                {'uses_custom_weather': False},
-                                format='json')
+        response = self.c.put('/mmw/modeling/scenarios/{}'
+                              .format(scenario['id']),
+                              scenario,
+                              format='json')
 
         self.assertEqual(response.status_code, 403)
 
-    def test_weather_patch_missing_param_400s(self):
-        project_id = self.create_private_project_with_weather_data()
+    def test_weather_put_invalid_param_400s(self):
+        scenario = self.create_private_scenario_with_weather_data()
+        scenario['weather_type'] = 'A_WRONG_VALUE'
 
-        response = self.c.patch(self.endpoint(project_id),
-                                {'XYZ': False},
```

Comment:
This use case is now handled by built-in DRF serializer validation, don't need to test in our code.